gradle-guide-error-prone/src/main/java/com/palantir/gradle/guide/errorprone/CallGraphBugChecker.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.guide.errorprone;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.graph.GraphBuilder;
+import com.google.common.graph.ImmutableGraph;
+import com.google.common.graph.MutableGraph;
+import com.google.common.graph.Traverser;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+public abstract class CallGraphBugChecker extends GradleGuideBugChecker {
+    /**
+     * A directed graph of "who is called by who" within this compilation unit (source file).
+     */
+    protected class MethodCallGraph {
+
+        /**
+         * Nodes are {@code MethodSymbol}s. There is an edge from {@code f} to {@code g} iff {@code f} is called by
+         * {@code g}.
+         */
+        @VisibleForTesting
+        protected final ImmutableGraph<MethodSymbol> callGraph;
+
+        public MethodCallGraph(VisitorState state) {
+            MutableGraph<MethodSymbol> mutableCallGraph =
+                    GraphBuilder.directed().allowsSelfLoops(true).build();
+
+            new SuppressibleTreePathScanner<Void, Optional<MethodSymbol>>(state) {
```

Comment:
Does this mean that a `@SuppressWarnings` will stop us from finding whole subgraphs because "on the way" something was suppressed? ie, here we would not even visit `anotherMethod` because `suppressed` has been suppressed?

```java
@TaskAction
void action() {
    suppressed();
}

@SuppressWarnings("IllegalMethodCalledDuringTaskExecution")
void suppressed() {
    anotherMethod();
}

void anotherMethod() {
    getProject();
}
```

Reply:
I think if we don't need to use `SuppressibleTreePathScanner` we don't need to make this a subclass of `BugChecker` and it can just be a static utility.

Reply:
We need some test for the `SuppressWarnings` cases at least

Reply:
Hmm good suggestion! Conveniently BugChecker offers the `isSuppressed` method, let's use that in the check itself, instead of using `SuppressibleTreePathScanner` here. 

This has allowed me to make the `MethodCallGraph` it's own class independent of the `BugChecker`. 

I've also added [a test](https://github.com/palantir/gradle-guide/pull/173/files#diff-29ad2989688f02530e9029c303ad7215d0af94dd532926de4a1627ab0113ecc1R225) for the case you described above.